### PR TITLE
ENH: increase coverage on common plugins

### PR DESCRIPTION
**Pull request by chenqi0805**
*Issue #, if available:*
#239 

*Description of changes:*
- increase test coverage on common plugins
- raise the enforcement to 90%
- delete unused API

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/prepper/StringPrepperTests.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public void testStringPrepperDefault() {
+        final StringPrepper stringPrepper = new StringPrepper(new PluginSetting(PLUGIN_NAME, new HashMap<>()));
+        final List<Record<String>> modifiedRecords = (List<Record<String>>) stringPrepper.execute(TEST_RECORDS);
+        stringPrepper.shutdown();
+
+        final List<String> modifiedRecordData = modifiedRecords.stream().map(Record::getData).collect(Collectors.toList());
+        final List<String> expectedRecordData = Arrays.asList(TEST_DATA_1.toUpperCase(), TEST_DATA_2);
+
+        Assert.assertTrue(modifiedRecordData.containsAll(expectedRecordData));
+        Assert.assertTrue(expectedRecordData.containsAll(modifiedRecordData));
```
Comment (wrijeff) [inline data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/prepper/StringPrepperTests.java:34]: Just `assertEquals()` should be sufficient here

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/prepper/StringPrepperTests.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StringPrepperTests {
+
+    private final String PLUGIN_NAME = "string_converter";
+
+    private final String TEST_DATA_1 = "data_prepper";
+    private final String TEST_DATA_2 = "STRING_CONVERTER";
```
Comment (wrijeff) [inline data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/prepper/StringPrepperTests.java:19]: nit - might name these something like UPPERCASE_TEST_STRING and LOWERCASE_TEST_STRING to make it more obvious that they have special properties (i.e. they aren't just random strings).

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    private static String PLUGIN_NAME = "stdout";
+
+    private final String TEST_DATA_1 = "data_prepper";
+    private final String TEST_DATA_2 = "stdout_sink";
+    private final Record<String> TEST_RECORD_1 = new Record<>(TEST_DATA_1);
+    private final Record<String> TEST_RECORD_2 = new Record<>(TEST_DATA_2);
+    private final List<Record<String>> TEST_RECORDS = Arrays.asList(TEST_RECORD_1, TEST_RECORD_2);
+
+    @Test
+    public void testSimple() {
```
Comment (wrijeff) [inline data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java:21]: I'm assuming there's nothing meaningful to test for this class?

File: data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/prepper/StringPrepper.java [hunk 4]
```diff
@@ -27,7 +29,7 @@
      * @param pluginSetting instance with metadata information from pipeline pluginSetting file.
      */
     public StringPrepper(final PluginSetting pluginSetting) {
-        this.upperCase = pluginSetting.getBooleanOrDefault("upper_case", true);
+        this.upperCase = pluginSetting.getBooleanOrDefault(UPPER_CASE, true);
```
Comment (wrijeff) [inline data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/prepper/StringPrepper.java:32]: 👍

**Review by wrijeff — APPROVED** [review]
👍 , minor comments

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java (hunk 3 above)
Comment (chenqi0805) [reply data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java:21]: None I can think of. Just make sure the API runs with no exception.

**Review by yadavcbala — APPROVED** [review]

**Pull request merged**